Source/lua/usertypes/items.cpp

```diff
@@ -0,0 +1,207 @@
+	    "RuneOfFire", SpellID::RuneOfFire,
+	    "RuneOfLight", SpellID::RuneOfLight,
+	    "RuneOfNova", SpellID::RuneOfNova,
+	    "RuneOfImmolation", SpellID::RuneOfImmolation,
+	    "RuneOfStone", SpellID::RuneOfStone,
+	    "Invalid", SpellID::Invalid);
+
+	lua.new_usertype<Item>("Item",
+	    "_iSeed", &Item::_iSeed,
+	    "_iCreateInfo", &Item::_iCreateInfo,
+	    "_iAnimFlag", &Item::_iAnimFlag,
+	    "position", &Item::position,
+	    "_iDelFlag", &Item::_iDelFlag,
+	    "selectionRegion", sol::property([](const Item &i) { return static_cast<int>(i.selectionRegion); }),
+	    "_iPostDraw", &Item::_iPostDraw,
+	    "_iIdentified", &Item::_iIdentified,
+	    "_iMagical", sol::property([](const Item &i) { return static_cast<int>(i._iMagical); }),
+	    "_iName", &Item::_iName,
+	    "_iIName", &Item::_iIName,
+	    "_iLoc", &Item::_iLoc,
+	    "_iClass", &Item::_iClass,
+	    "_iCurs", &Item::_iCurs,
+	    "_ivalue", &Item::_ivalue,
+	    "_iIvalue", &Item::_iIvalue,
+	    "_iMinDam", &Item::_iMinDam,
+	    "_iMaxDam", &Item::_iMaxDam,
+	    "_iAC", &Item::_iAC,
+	    "_iFlags", sol::property([](const Item &i) { return static_cast<int>(i._iFlags); }),
+	    "_iMiscId", &Item::_iMiscId,
+	    "_iSpell", &Item::_iSpell,
+	    "IDidx", &Item::IDidx,
+	    "_iCharges", &Item::_iCharges,
+	    "_iMaxCharges", &Item::_iMaxCharges,
+	    "_iDurability", &Item::_iDurability,
+	    "_iMaxDur", &Item::_iMaxDur,
+	    "_iPLDam", &Item::_iPLDam,
+	    "_iPLToHit", &Item::_iPLToHit,
+	    "_iPLAC", &Item::_iPLAC,
+	    "_iPLStr", &Item::_iPLStr,
+	    "_iPLMag", &Item::_iPLMag,
+	    "_iPLDex", &Item::_iPLDex,
+	    "_iPLVit", &Item::_iPLVit,
+	    "_iPLFR", &Item::_iPLFR,
+	    "_iPLLR", &Item::_iPLLR,
+	    "_iPLMR", &Item::_iPLMR,
+	    "_iPLMana", &Item::_iPLMana,
+	    "_iPLHP", &Item::_iPLHP,
+	    "_iPLDamMod", &Item::_iPLDamMod,
+	    "_iPLGetHit", &Item::_iPLGetHit,
+	    "_iPLLight", &Item::_iPLLight,
+	    "_iSplLvlAdd", &Item::_iSplLvlAdd,
+	    "_iRequest", &Item::_iRequest,
+	    "_iUid", &Item::_iUid,
+	    "_iFMinDam", &Item::_iFMinDam,
+	    "_iFMaxDam", &Item::_iFMaxDam,
+	    "_iLMinDam", &Item::_iLMinDam,
+	    "_iLMaxDam", &Item::_iLMaxDam,
+	    "_iPLEnAc", &Item::_iPLEnAc,
+	    "_iPrePower", sol::property([](const Item &i) { return static_cast<int>(i._iPrePower); }),
+	    "_iSufPower", sol::property([](const Item &i) { return static_cast<int>(i._iSufPower); }),
+	    "_iVAdd1", &Item::_iVAdd1,
+	    "_iVMult1", &Item::_iVMult1,
+	    "_iVAdd2", &Item::_iVAdd2,
+	    "_iVMult2", &Item::_iVMult2,
+	    "_iMinStr", &Item::_iMinStr,
+	    "_iMinMag", &Item::_iMinMag,
+	    "_iMinDex", &Item::_iMinDex,
+	    "_iStatFlag", &Item::_iStatFlag,
+	    "_iDamAcFlags", sol::property([](const Item &i) { return ItemSpecialEffectHfToString(i._iDamAcFlags); }),
+	    "dwBuff", &Item::dwBuff);
+}
+
+sol::object CreateItemObject(const Item &item)
+{
+	sol::state &L = GetLuaState();
+
+	if (!L["Item"].valid())
+		RegisterItemUsertype(L);

```

Comment:
This should probably be called when setting up the `items` module?

Reply:
Will open a new smaller PR for setting this up like you did with the player usertype.